File: creational/abstract_factory/ruby/lib/creator/windows_widget_factory.rb

require_relative '../product/windows_button'
require_relative '../product/windows_checkbox'
require_relative './widget_factory'


class WindowsWidgetFactory < WidgetFactory
  def create_button
    WindowsButton.new
  end

  def create_checkbox
    WindowsCheckbox.new
  end
end

File: creational/abstract_factory/java/src/creator/WindowsWidgetFactory.java

package creator;

import product.Button;
import product.CheckBox;
import product.WindowsButton;
import product.WindowsCheckBox;


public class WindowsWidgetFactory implements WidgetFactory {
    @Override
    public Button createButton() {
        return new WindowsButton();
    }

    @Override
    public CheckBox createCheckBox() {
        return new WindowsCheckBox();
    }
}
